spotlight-client/README.md

```diff
@@ -10,6 +10,8 @@ The site consumes views of aggregate population data produced by the Recidiviz d
 
 While Spotlight is developed and deployed as a single multi-tenant website, it is primarily consumed as separate single-tenant experiences, under `.gov` subdomains owned by our state partners (e.g. [dashboard.docr.nd.gov](https://dashboard.docr.nd.gov)). To keep our infrastructure simple, this "tenant lock" is implemented in application logic within the data models, based on the URL hostname at runtime. This is why the multi-tenant "homepage" is so plain; it is really only used internally, for convenience, in development and staging environments.
 
+We also lock the staging environment to a single tenant depending on how the logged-in user is configured. In the staging environment, a user logs in to view the site, and if their Auth0 account has a `state_code` set in their `app_metadata`, then they will be locked-in to the tenant that correspons with that `state_code`. This is so we can share a fully-functional but private version of the app with contacts of that associated agency, without exposing data to state actors that do not have permission to view other states’ data. The `state_code` can be set by going to the `recidiviz-spotlight-staging` tenant in Auth0, looking up the user and adding the `state_code` to their `app_metadata`.
```

Comment:
Do we really plan on manually adding the state_code for every user? 

What happens the first time they log in? The user account will be created when they register, and once they then verify their account they will have full access to the unlocked app until we are able to go in and manually set the state_code for that user. 

You may want to take a look at the [pre-registration-setup](https://github.com/Recidiviz/pulse-dashboard/blob/main/src/auth0/actions/01-pre-registration-setup.js) action in pulse-dashboard to see how we are setting their state_code when the user registers - this ensures that they will ALWAYS have a state_code set, which means they will never see the unlocked version. I assume users in spotlight staging will create accounts and not be on SSO? If they are on SSO (which I highly doubt but want to double check) we will need to do it differently.

Reply:
I'm unfamiliar with how onboarding users work, whether it's a manual process of creating accounts in Auth0 + setting the state_code or if a user can register themselves. 

That's cool that we can have a pre-registration-setup action to configure users to the right `state_code` when users register. I'm thinking I should use something like this to assign users to a `state_code` when they register. Perhaps only show all states if the state_code is `recidiviz`, just to be safe?

Reply:
For those users who have already registered, we can manually set their state_code and they should be able to continue using the app - the state_code in the metadata will not disappear. Adding the pre-registration action will only affect new users.

Reply:
> * I think this could be done in a separate PR, up to you guys.

Given that this is all public data I think this is probably fine? If this was in any other app that is not public data, it would need to be done in the same PR because not doing it allows users access to data that they should not see. What is our data sharing agreement like with the Spotlight states for the staging data?

Reply:
I just updated this PR to not show any tenants if the `state_code` is not set, and to show all tenants if the `state_code` is set to `recidiviz`. 

I might as well update this PR to include a pre-registration flow to add the `state_code` to users when they sign up? Maybe I can add a login step to add it to users who log in and don't have a `state_code` set?

Reply:
Another idea: instead of setting the state_code, we lock users based on their email domains. So @recidiviz.org users will always see all states, nd.gov users will only see North Dakota, but I don't know if this system works for us or if it's too inflexible.

Reply:
@hobuobi let me know if this user flow makes sense

Reply:
Just chatted with humphrey, I'll look into creating a pre-registration flow script to assign new users to a state_code, and also see if it's easy to assign pre-existing users a state_code when they login if they're missing one.

Reply:
Just added a post-login script to automatically set the state_code for nd pa and recidiviz users. Note we will have to edit both our existing domain whitelist as well as this post-login script to support a new state. Should that be documented somewhere?